tests/prefetch_related/tests.py

```diff
@@ -306,6 +306,28 @@ def test_filter_deferred(self):
             list(Book.objects.prefetch_related('authors'))
             self.assertEqual(add_q_mock.call_count, 1)
 
+    def test_foreignkey_forward_filter_deferred(self):
+        add_q = Query.add_q
+        with mock.patch.object(
+            Query,
+            'add_q',
+            autospec=True,
+            side_effect=lambda self, q: add_q(self, q),
+        ) as add_q_mock:
+            list(Book.objects.prefetch_related('first_book'))
+            self.assertEqual(add_q_mock.call_count, 1)
+
+    def test_foreignkey_reverse_filter_deferred(self):
+        add_q = Query.add_q
+        with mock.patch.object(
+            Query,
+            'add_q',
+            autospec=True,
+            side_effect=lambda self, q: add_q(self, q),
+        ) as add_q_mock:
+            list(Book.objects.prefetch_related('first_time_authors'))
+            self.assertEqual(add_q_mock.call_count, 1)
```

Comment:
We can use `self.subTest()` and add test for a m2o relation to the `test_filter_deferred`, e.g.
```python
    def test_filter_deferred(self):
        """
        Related filtering of prefetched querysets is deferred until necessary.
        """
        add_q = Query.add_q
        for relation in ['authors', 'first_time_authors']:
            with self.subTest(relation=relation):
                with mock.patch.object(
                    Query,
                    'add_q',
                    autospec=True,
                    side_effect=lambda self, q: add_q(self, q),
                ) as add_q_mock:
                    list(Book.objects.prefetch_related(relation))
                    self.assertEqual(add_q_mock.call_count, 1)
```

Reply:
Thanks for the heads up - done in 776c6261751d7b7c30b76fe8dcc67ffb1c244cae